internal/manager/run.go

```diff
@@ -229,3 +238,9 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 	setupLog.Info("Starting manager")
 	return mgr.Start(ctx)
 }
+
+func isControllerNameValid(controllerName string) bool {
+	// https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/shared_types.go#L451-L466
+	re := regexp.MustCompile(`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*\/[A-Za-z0-9\/\-._~%!$&'()*+,;=:]+$`)
+	return re.Match([]byte(controllerName))
+}
```

Comment:
This whole function _might_ be reasonable to add as an upstream utility if you felt so inclined. Upstream has recently started allowing "utility functions" for the Go lib.

Reply:
Sure, I'll open an issue upstream about this. Can we keep this one in the meantime?

Reply:
Yep